openfisca_core/entities/_core_entity.py

```diff
@@ -90,16 +126,47 @@ def check_variable_defined_for_entity(self, variable_name: t.VariableName) -> No
         Args:
             variable_name: The ``Variable`` to be found.
 
-        Returns:
-            Variable: When the ``Variable`` exists.
-            None: When the :attr:`_tax_benefit_system` is not set.
-
         Raises:
             ValueError: When the ``Variable`` exists but is defined
                 for another ``Entity``.
 
-        """
+        Examples:
+            >>> from openfisca_core import (
+            ...     entities,
+            ...     periods,
+            ...     taxbenefitsystems,
+            ...     variables,
+            ... )
+
+            >>> this = entities.SingleEntity("this", "", "", "")
+            >>> that = entities.SingleEntity("that", "", "", "")
+            >>> tax_benefit_system = taxbenefitsystems.TaxBenefitSystem([that])
+            >>> this.set_tax_benefit_system(tax_benefit_system)
+
+            >>> this.check_variable_defined_for_entity("tax")
+            Traceback (most recent call last):
+            VariableNotFoundError: You tried to calculate or to set a value...
+
+            >>> class tax(variables.Variable):
+            ...     definition_period = periods.WEEK
+            ...     value_type = int
+            ...     entity = that
```

Comment:
Same as before.
Why using different entities, `this` and `that`  here ?

Reply:
Same answer as above.